include/small/matras.h

```diff
@@ -138,8 +138,26 @@ typedef uint32_t matras_id_t;
  * of size M). Is allowed to return NULL, but is not allowed
  * to throw an exception
  */
-typedef void *(*matras_alloc_func)(void *ctx);
-typedef void (*matras_free_func)(void *ctx, void *ptr);
+struct matras_allocator;
+typedef void *(*matras_alloc_func)(struct matras_allocator *allocator);
+typedef void (*matras_free_func)(struct matras_allocator *allocator, void *ptr);
+
+/**
+ * Matras extent allocator. Wraps the external allocator and allows to
+ * reserve the speciied amount of extents for future use.
+ */
+struct matras_allocator {
+	/** Single extent size. */
+	size_t extent_size;
+	/** External extent allocator */
+	matras_alloc_func alloc_func;
+	/** External extent deallocator */
+	matras_free_func free_func;
+	/** Reserved extents (linked list). */
+	void *reserved_extents;
+	/** Number of reserved extents. */
+	unsigned num_reserved_extents;
```

Comment:
Nit: Please don't use `unsigned` for counters. Use `int` and `assert` to ensure it stays positive. [Here](https://google.github.io/styleguide/cppguide.html#Integer_Types) you can find a brief rationale.

Reply:
Changes the field type.